src/translator/quality_estimator.h

```diff
@@ -0,0 +1,209 @@
+
+    /// Return data value given a row and col position
+    /// @param [in] row: row position
+    /// @param [in] col: col position
+    const float &at(const size_t row, const size_t col) const;
+    float &at(const size_t row, const size_t col);
+
+   private:
+    std::vector<float> data_;
+  };
+  /// Logistic Regressor constructor. It creates a LR model that fits proper for the QualityEstimator use.
+  ///
+  ///
+  /// @param [in] scale: Array of stds and means that can be used to apply standard scaling in the features
+  /// @param [in] coefficients: coefficient values of linear part of LR model
+  /// @param [in] intercept: intercept value of the linear part of LR model
+  LogisticRegressorQualityEstimator(Scale &&scale, std::vector<float> &&coefficients, const float intercept);
+
+  /// Move constructor
+  LogisticRegressorQualityEstimator(LogisticRegressorQualityEstimator &&other);
+
+  /// Binary file parser which came from AlignedMemory
+  /// It's expected from AlignedMemory the following structure:
+  /// - -a header with the number of parameters dimensions
+  /// - -a vector of standard deviations of features
+  /// - -a vector of means of features
+  /// - -a vector of coefficients
+  /// - -a intercept value
+  static LogisticRegressorQualityEstimator fromAlignedMemory(const AlignedMemory &alignedMemory);
+  AlignedMemory toAlignedMemory() const;
+
+  void computeQualityScores(const Histories &histories, Response &response) const override;
+
+  /// Given an input matrix $\mathbf{X}$, the usual Logistic Regression calculus can be seen as the following:
+  ///
+  /// 1) Standardize it, returning in $\mathbf{Z} = \frac{(\mathbf{X}-\mu)}{\sigma}$, where $\mu$ stands for the mean
+  /// vector and $\sigma$ represents the standard deviation
+  ///
+  /// 2) Then, we apply $\sum_{i=1}^{D}{ w_i z_i}$, where $D$ is the dimension (i.e. the number of features) and $w$ is
+  /// the model vector with learnt weights
+  ///
+  /// 3) We apply the sigmoid function to the result
+  ///
+  /// Notice, however, that for the first two steps we can do the following:
+  ///
+  /// \begin{align*}
+  /// \sum_{i=1}^{D}{ w_i z_i} &= \mathbf{w^T}\left(\mathbf{\sigma^{-1}} \odot (\mathbf{x} - \mathbf{\mu})\right) \text{
+  /// //
+  /// we are just vectoring step 1}\\
+  ///      &= \sum_{i=1}^{D}{\sigma_i^{-1} w_i (x_i - \mu_i)} \\
+  ///      &= \sum_{i=1}^{D}{\sigma_i^{-1} w_ix_i - \sigma_i^{-1} w_i \mu_i} \\
+  ///      &= \sum_{i=1}^{D}{\left(\sigma_i^{-1} w_i\right)x_i - \left(\sigma_i^{-1} w_i \mu_i\right)}
+  /// \end{align*}
+  /// Then, $(\sigma_i^{-1} w_i \mu_i)$ can be precomputed without any dependence on inference data. This is done by the
+  /// variable $\textit{constantFactor_}$ and $\textit{intercept_}$ in the code.
+  ///
+  /// For a better reading please refer to: http://mathb.in/63082
+  ///
+  ///
+  /// @param [in] features: A Matrix struct of features
+  std::vector<float> predict(const Matrix &features) const;
```

Comment:
What exactly does `feature` mean in this context? It would be helpful to document it briefly.

Reply:
Please see if you agree with c762e3b